src/System.IO.FileSystem/src/System/IO/FileSystem.Unix.cs

```diff
@@ -40,6 +69,9 @@ public static void Decrypt(string path)
 
         private static void LinkOrCopyFile (string sourceFullPath, string destFullPath)
         {
+            if (CopyDanglingSymlink(sourceFullPath, destFullPath))
+                return;
```

Comment:
Can we do this instead after the attempt to call `Link` below rather than before?  My assumption is that Link will fail with a dangling link, in which case we can do this CopyDanglingSymlink as part of the slow fallback path below rather than always paying for these additional costs upfront?